### PR TITLE
Upgrade argocd-operator version in go.mod

### PR DESCRIPTION
**Pull request by chetan-rns**
Upgrade argocd-operator version in go.mod to include https://github.com/argoproj-labs/argocd-operator/pull/643

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
would you also like to compare the manifest files as part of the same PR ?

**Comment by chetan-rns** [comment]
@iam-veeramalla Thanks for reminding me. I've updated the CRDs in the same PR.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/redhat-developer/gitops-operator
 go 1.16
```
Comment (wtam2018) [inline go.mod:3]: Are we upgrading go version here?

**Review by wtam2018 — COMMENTED** [review]
How about the Dockefile?

File: go.mod (hunk 1 above)
Comment (chetan-rns) [reply go.mod:3]: There is a different PR for upgrading go version https://github.com/redhat-developer/gitops-operator/pull/307

**Comment by wtam2018** [comment]
Also, in https://github.com/argoproj-labs/argocd-operator/pull/643, we should update the go version in .github/workflows.  @chetan-rns

**Comment by chetan-rns** [comment]
> Also, in [argoproj-labs/argocd-operator#643](https://github.com/argoproj-labs/argocd-operator/pull/643), we should update the go version in .github/workflows. @chetan-rns

It will be updated in a different PR https://github.com/argoproj-labs/argocd-operator/pull/624

**Comment by chetan-rns** [comment]
@wtam2018 @iam-veeramalla  @jopit CI passed. Can we merge this PR?

**Pull request merged**